File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.java
class Solution {
    public int countCollisions (String dir) {
        final int n = dir.length ();
        if (n == 1) return 0;
        int l = 0, r = n - 1;
        while (l < r && dir.charAt (l) == 'L') l++;
        while (l < r && dir.charAt (r) == 'R') r--;
        if (l >= r) return 0;
        int col = 0;
        for (; l <= r; l++)
           col += (dir.charAt (l) != 'S' ? 1 : 0);
        return col;  
    }
}

File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.c
#pragma GCC optimize ("O3, unroll-loops")

int countCollisions (char* dir) {
    const int n = strlen (dir);
    if (n == 1) return 0;
    int l = 0, r = n - 1;
    while (dir [l] == 'L') l++;
    while (l < r && dir [r] == 'R') r--;
    if (l >= r) return 0;
    int col = 0;
    for(; l <= r; l++) col += dir [l] != 'S';
    return col;
}